diskutil.py

```diff
@@ -289,9 +289,9 @@ def getDiskBlockSize(dev):
         dev = '/dev/' + dev
     if isDeviceMapperNode(dev):
         return getDiskBlockSize(getDeviceSlaves(dev)[0])
-    if dev.startswith("/dev/"):
-        dev = re.match("/dev/(.*)", dev).group(1)
-    dev = dev.replace("/", "!")
+
+    realpath = os.path.realpath(dev)
+    dev = os.path.basename(realpath)
```

Comment:
Yes, the code removing `/dev/` was redundant giving previous lines, however this not explain this change. The same code you are removing and change is present on the previous function. I think this code is dealing with devices having path separator in them. How your code is dealing with them? It looks like you are removing the support of such devices instead.

Reply:
Yes, by using realpath to reduce that down to the actual device rather than the symlink, which should not have path separators in its name.

Reply:
Some device name have path separators in their names